### PR TITLE
Fixes for Lie groups and homogeneous spaces

### PR DESCRIPTION
**Pull request by vabor112**

### CODE REVIEW

File: geometric_kernels/spaces/lie_groups.py [hunk 1]
```diff
@@ -84,25 +87,25 @@ def _addition_theorem(self, X: B.Numeric, X2: B.Numeric, **parameters) -> B.Nume
         diff = self._difference(X, X2)
         torus_repr_diff = self._torus_representative(diff)
         values = [
-            degree**2 * chi(torus_repr_diff)[..., None]  # [N1, N2, 1]
-            for chi, degree in zip(self._characters, self._dimensions)
+            repr_dim * B.real(chi(torus_repr_diff)[..., None])  # [a, b, 1]
```
Comment (stoprightthere) [inline geometric_kernels/spaces/lie_groups.py:90]: I think we use capital letters N, M etc for shapes.

File: geometric_kernels/spaces/lie_groups.py [hunk 2]
```diff
@@ -84,25 +87,25 @@ def _addition_theorem(self, X: B.Numeric, X2: B.Numeric, **parameters) -> B.Nume
...
+            for chi, repr_dim in zip(self._characters, self._dimensions)
         ]
-        return B.concat(*values, axis=-1)  # [N, N2, L]
+        return B.concat(*values, axis=-1)  # [a, b, L]
 
     def _addition_theorem_diag(self, X: B.Numeric, **parameters) -> B.Numeric:
         """
         Returns the sum of eigenfunctions on a level for which we have a simplified expression
-        :param X: [N, D]
+        :param X: [a, n, n]
```
Comment (stoprightthere) [inline geometric_kernels/spaces/lie_groups.py:98]: I think the convention is to use "N" (and similar) for the number of points and "D" or "D, D" for dimensions.

File: geometric_kernels/spaces/lie_groups.py [hunk 3]
```diff
@@ -144,7 +142,15 @@ def __call__(self, gammas):
...
 class MatrixLieGroup(DiscreteSpectrumSpace):
     r"""
-    A base class for Lie groups represented as matrices. The group operation is the matrix multiplication, and the inverse is the matrix inverse.
+    A base class for Lie groups of matrices, subgroups of the general linear
+    group GL(n).
+
+    The group operation is the standard matrix multiplication, and the group
+    inverse is the standard matrix inverse. Despite this, we make the
+    subclasses implement their own multiplication/inverse routines, because
+    in special cases it can typically be implemented much more efficient.
```
Comment (stoprightthere) [inline geometric_kernels/spaces/lie_groups.py:151]:
```suggestion
    in special cases they can typically be implemented much more efficiently.
```

File: geometric_kernels/spaces/lie_groups.py [hunk 4]
```diff
@@ -144,7 +142,15 @@ def __call__(self, gammas):
 
 class MatrixLieGroup(DiscreteSpectrumSpace):
     r"""
-    A base class for Lie groups represented as matrices. The group operation is the matrix multiplication, and the inverse is the matrix inverse.
+    A base class for Lie groups of matrices, subgroups of the general linear
+    group GL(n).
+
+    The group operation is the standard matrix multiplication, and the group
+    inverse is the standard matrix inverse. Despite this, we make the
```
Comment (stoprightthere) [inline geometric_kernels/spaces/lie_groups.py:149]: "we make the subclasses implement their own multiplication/inverse routines" -> "we defer the implementation of the inverse to the subclasses".

We don't have a special multiplication, do we? Only the inverse is overloaded.

**Review by stoprightthere — CHANGES_REQUESTED** [review]
Good work, see some minor comments!

File: geometric_kernels/spaces/lie_groups.py (hunk 3 above)
Comment (vabor112) [reply geometric_kernels/spaces/lie_groups.py]: Done.

File: geometric_kernels/spaces/lie_groups.py (hunk 4 above)
Comment (vabor112) [reply geometric_kernels/spaces/lie_groups.py:228]: Done.

File: geometric_kernels/spaces/lie_groups.py (hunk 2 above)
Comment (vabor112) [reply geometric_kernels/spaces/lie_groups.py]: Done.

File: geometric_kernels/spaces/lie_groups.py [hunk 5]
```diff
@@ -6,12 +6,53 @@
...
+    :param n:
+    The order of the Lie group, e.g. for SO(5) this is 5, for SU(3) this is 3.
+    :param num_levels:
+    The number of levels used for kernel approximation. Here, each level
+    corresponds to an irreducible unitary representation of the group.
+    :param compute_characters:
+    Whether or not to actually compute the *characters*. Setting this parameter
+    to False might make sense if you do not care about eigenfunctions (or sums
+    of outer products thereof), but care about eigenvalues, dimensions of
+    irreducible unitary representations, etc. Dafaults to True.
```
Comment (stoprightthere) [inline geometric_kernels/spaces/lie_groups.py:34]:
```suggestion
    irreducible unitary representations, etc. Defaults to True.
```

**Review by stoprightthere — APPROVED** [review]

**Pull request merged**